<!-- 
Launches SDK driver and transforms publisher.
Depends on `manager.launch`.
-->
<launch>
  <arg name="capture_loop" default="true" doc="Enable cpture replay looping."/>
  <arg name="capture_path" default="" doc="Capture replay PCAP file path."/>
  <arg name="control_flags" default="0" doc="SDK control flags."/>
  <arg name="frame_mode" default="CYCLE" doc="SDK frame mode (STREAMING, COVER, CYCLE)."/>
  <arg name="manager_name" default="cepton_manager" doc="Nodelet manager node name."/>
  <arg name="parent_frame_id" default="cepton" doc="Parent transform frame."/>
  <arg name="settings_dir" default="" doc="Sensor settings directory."/>

  <node pkg="nodelet" type="nodelet" name="cepton_driver" required="true" args="load cepton_ros/DriverNodelet $(arg manager_name)" output="screen">
    <param name="capture_path" value="$(arg capture_path)"/>
    <param name="capture_loop" value="$(arg capture_loop)"/>
    <param name="control_flags" value="$(arg control_flags)"/>
    <param name="frame_mode" value="$(arg frame_mode)"/>
    <param name="parent_frame_id" value="$(arg parent_frame_id)"/>
    <param name="settings_dir" value="$(arg settings_dir)"/>
  </node>
</launch>
